lib/fog/aws/requests/dynamodb/query.rb

```diff
@@ -25,15 +23,17 @@ class Real
         #     * 'Count'<~Integer> - number of items in response
         #     * 'Items'<~Array> - array of items returned
         #     * 'LastEvaluatedKey'<~Hash> - last key scanned, can be passed to ExclusiveStartKey for pagination
-        def query(table_name, hash_key, options = {})
+        #
+        # See DynamoDB Documentation: http://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_Query.html
+        #
+        def query(table_name, options = {})
```

Comment:
The previous API required a HashKeyValue argument, which contained: `{:S => 1}`.
See http://docs.aws.amazon.com/amazondynamodb/latest/developerguide/API_Query_v20111205.html

DynamoDB would take the id and compare the provided value with it.
The new API requires to do:

```
:KeyConditionExpression => 'id = :id'
```

And unless the DynamoDB documentation is wrong, it doesn't appear `HashKeyValue` is still supported.
Transforming the former hash to the new value seems difficult, as we don't know the name of the primary key.